common/src/api/external/mod.rs

```diff
@@ -2894,39 +2894,6 @@ pub struct SwitchPortAddressConfig {
     pub interface_name: String,
 }
 
-/// Opaque object representing link state. The contents of this object are not
-/// yet stable.
-#[derive(Clone, Debug, Deserialize, Serialize)]
-pub struct SwitchLinkState {
```

Comment:
The bulk of this PR is moving this type to `nexus/types/src/external_api/shared.rs`